docs/providers.rst

```diff
@@ -211,6 +218,72 @@ WebsocketProvider
         >>> from web3 import Web3
         >>> w3 = Web3(Web3.WebsocketProvider("ws://127.0.0.1:8546", websocket_timeout=60))
 
+
+WebsocketProviderV2 (beta)
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. py:class:: web3.providers.websocket.WebsocketProviderV2(endpoint_uri, websocket_kwargs, call_timeout)
+
+    This provider handles interactions with an WS or WSS based JSON-RPC server.
+
+    * ``endpoint_uri`` should be the full URI to the RPC endpoint such as
+      ``'ws://localhost:8546'``.
+    * ``websocket_kwargs`` this should be a dictionary of keyword arguments which
+      will be passed onto the ws/wss websocket connection.
+    * ``call_timeout`` is the timeout in seconds, used when receiving or sending data
+      over the connection. Defaults to ``None`` (no timeout).
+
+    .. code-block:: python
+
+        >>> import asyncio
+        >>> from web3 import Web3
+        >>> from web3.providers import WebsocketProviderV2
+
+        >>> LOG = True  # toggle debug logging
+        >>> if LOG:
+        >>>     import logging
+        >>>     logger = logging.getLogger("web3.providers.WebsocketProviderV2")
+        >>>     logger.setLevel(logging.DEBUG)
+        >>>     logger.addHandler(logging.StreamHandler())
```

Comment:
nit
```suggestion
        ...     import logging
        ...     logger = logging.getLogger("web3.providers.WebsocketProviderV2")
        ...     logger.setLevel(logging.DEBUG)
        ...     logger.addHandler(logging.StreamHandler())
```

Reply:
As far as I've ever used the ellipsis in code blocks it's a continuation of the same statement. These are all separate lines. Is this common usage?

Reply:
I see it as a single statement because it's all within a single `if`.

Reply:
Gotcha. I'm good with it 👍🏼. I'll pull this change in.